plugins/inputs/execd/shim/goshim.go

```diff
@@ -0,0 +1,281 @@
+package shim
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/agent"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/serializers/influx"
+)
+
+type empty struct{}
+
+var (
+	gatherPromptChans []chan empty
+	stdout            io.Writer = os.Stdout
+	stdin             io.Reader = os.Stdin
+)
+
+const (
+	// PollIntervalDisabled is used to indicate that you want to disable polling,
+	// as opposed to duration 0 meaning poll constantly.
+	PollIntervalDisabled = time.Duration(0)
+)
+
+type Shim struct {
+	Inputs   []telegraf.Input
+	metricCh chan telegraf.Metric
+}
+
+func New() *Shim {
+	return &Shim{
+		metricCh: make(chan telegraf.Metric, 1),
+	}
+}
+
+// AddInput adds the input to the shim. Later calls to Run() will run this input.
+func (s *Shim) AddInput(input telegraf.Input) error {
+	if p, ok := input.(telegraf.Initializer); ok {
+		err := p.Init()
+		if err != nil {
+			return fmt.Errorf("failed to init input: %s", err)
+		}
+	}
+
+	s.Inputs = append(s.Inputs, input)
+	return nil
+}
+
+// AddInputs adds multiple inputs to the shim. Later calls to Run() will run these.
+func (s *Shim) AddInputs(newInputs []telegraf.Input) error {
+	for _, inp := range newInputs {
+		if err := s.AddInput(inp); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Run the input plugins..
+func (s *Shim) Run(pollInterval time.Duration) error {
+	wg := sync.WaitGroup{}
+	quit := make(chan os.Signal, 1)
+	signal.Notify(quit, syscall.SIGINT, syscall.SIGTERM)
+
+	collectMetricsPrompt := make(chan os.Signal, 1)
+	listenForCollectMetricsSignals(collectMetricsPrompt)
+
+	wg.Add(1) // wait for the metric channel to close
+	metricCh := make(chan telegraf.Metric, 1)
```

Comment:
Is this different from `Shim.metricCh`?

Reply:
Ah, I started to move it but didn't end up needing it. Removing the struct copy.